### PR TITLE
feat: Github Actions workflows to automate initial validation of customer Github issues

### PR DESCRIPTION
**Pull request by daphne-sfdc**
### What does this PR do?
Two new Github Actions workflows – **Validate New Issues** and **Validate Updated Issues** that read the content of new Github issues.
- Feature requests submitted using the feature request template are automatically labeled as `type:enhancements`.
- Bug reports are checked for all the required basic info in the bug report template and tagged as `validated` if all required info is present.
- If the required info is missing, the workflows post a comment to ask for the missing info, and then tag the issue with `more information required`.
- If the customer provides an outdated version of the extensions, CLI, or VSCode, the workflows post a comment with instructions to update to the latest version, and then tag the issue with `more information required`.

### What issues does this PR fix or reference?
@W-15525741@

### Functionality Before
The support person has to read through each Github issue and manually ask the customer for more info if the customer did not completely fill out the bug report template.

### Functionality After
The initial validation is done by the new **Validate New Issues** and **Validate Updated Issues** Github Actions workflows, and the support person only needs to look at issues that the workflows have tagged as `validated`.

Notes here: https://salesforce.quip.com/2ralAJ3bZkq5

### CODE REVIEW

**Review by CristiCanizales — CHANGES_REQUESTED** [review]
This is great stuff Daphne! Thanks for all of this. I just have a couple of questions, suggestions and one concern. In summary:
- Do we need all the stuff (commented and uncommented code) for local testing? I just don't think we'll be doing local testing at all for this
- Do users need latest CLI version or just a valid one?
- Please update the date in all the copyrights and the name/author in workflows.
- I feel we're encouraging users too much to create cases, and that would involve SLA's and all that 🥵

File: .github/actions/validate-issue/lib/src/nodeVersions.d.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export declare const isAnyVersionValid: (currentDate: Date) => (versions: string[]) => Promise<boolean>;
+//# sourceMappingURL=nodeVersions.d.ts.map
```
[the remark this answers is not in the document]
Comment (daphne-sfdc) [reply .github/actions/validate-issue/lib/src/nodeVersions.d.ts:2]: `//# sourceMappingURL` in a \*.d.ts file is a source mapping URL comment. It's used to specify a source map file for the TypeScript declaration file (*.d.ts).  Thus, the comment necessary in this case and should not be removed.

File: .github/workflows/validateUpdatedIssues.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        run: npm install
+      - name: issue comment
+        id: issue_comment
+        uses: ./.github/actions/new-issue
+        with:
+          repo-token: ${{ secrets.IDEE_GH_TOKEN }}
+          message: >
+            We have determined that the issue you reported exists in code owned by another team that uses only the official support channels.
+            To ensure that your issue is addressed, open an official [Salesforce customer support](https://help.salesforce.com/s/) ticket with a link to this issue.
+            We encourage anyone experiencing this issue to do the same to increase the priority. We will keep this issue open for the community to collaborate on.
```
[the remark this answers is not in the document]
Comment (daphne-sfdc) [reply .github/workflows/validateUpdatedIssues.yml:76]: We only have 4 issues total that are owned by another team, and 2 of them are closed.  I think we should keep issues open for tracking purposes. https://github.com/forcedotcom/salesforcedx-vscode/issues?q=is%3Aissue+label%3A%22status%3Aowned+by+another+team%22

File: .github/actions/check-feature-request/lib/src/index.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+Object.defineProperty(exports, "__esModule", { value: true });
+const core_1 = require("@actions/core");
+const github_1 = require("@actions/github");
+async function run() {
+    try {
+        // Uncomment for local testing
+        // const issue = JSON.parse(getFile("../mock/sample-context.json"));
```
Comment (CristiCanizales) [inline .github/actions/check-feature-request/lib/src/index.js:14]:
```suggestion
```

File: .github/actions/check-feature-request/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { getInput, setFailed } from "@actions/core";
+import { context, getOctokit } from "@actions/github";
+
+async function run() {
+  try {
+    // Uncomment for local testing
+    // const issue = JSON.parse(getFile("../mock/sample-context.json"));
```
Comment (CristiCanizales) [inline .github/actions/check-feature-request/src/index.ts:14]:
```suggestion
```

File: .github/actions/validate-issue/lib/src/index.js [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+                    postComment(message);
+                    provideVersionAlreadyRequested = true;
+                    addLabel("missing required information");
+                }
+                lastWorkingVersionValid = false;
+            }
+            // *** The below is the check for CLI version, code reused from CLI Team's repo ***
+            const sfVersionRegex = /@salesforce\/cli\/([0-9]+.[0-9]+.[0-9]+(-[a-zA-Z0-9]+.[0-9]+)?)/g;
+            const sfdxVersionRegex = /sfdx-cli\/([0-9]+.[0-9]+.[0-9]+(-[a-zA-Z0-9]+.[0-9]+)?)/g;
+            // const pluginVersionsRegex = /pluginVersions|Plugin Version:/;
```
Comment (CristiCanizales) [inline .github/actions/validate-issue/lib/src/index.js:193]:
```suggestion
```

File: .github/actions/validate-issue/mock/sample-context.json
Comment (CristiCanizales) [inline .github/actions/validate-issue/mock/sample-context.json:1]: do we need this file?

File: .github/actions/validate-issue/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,431 @@
...
+          addLabel("missing required information");
+        }
+        lastWorkingVersionValid = false;
+      }
+
+      // *** The below is the check for CLI version, code reused from CLI Team's repo ***
+
+      const sfVersionRegex = /@salesforce\/cli\/([0-9]+.[0-9]+.[0-9]+(-[a-zA-Z0-9]+.[0-9]+)?)/g;
+      const sfdxVersionRegex = /sfdx-cli\/([0-9]+.[0-9]+.[0-9]+(-[a-zA-Z0-9]+.[0-9]+)?)/g;
+      // const pluginVersionsRegex = /pluginVersions|Plugin Version:/;
```
Comment (CristiCanizales) [inline .github/actions/validate-issue/src/index.ts:230]:
```suggestion
```

File: .github/actions/validate-issue/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,431 @@
...
+          console.log(
+            `Cannot remove label '${label}' since it was not applied`
+          );
+          return;
+        }
+        throw error;
+      }
+    }
+
+    function getLatestCliVersion(plugin: string) {
```
Comment (CristiCanizales) [inline .github/actions/validate-issue/src/index.ts:392]: are we using this? If not, please remove it

**Review by CristiCanizales — CHANGES_REQUESTED** [review]

**Review by CristiCanizales — APPROVED** [review]
Looks good! Thanks for addressing all my comments!

:shipit:

**Pull request merged**